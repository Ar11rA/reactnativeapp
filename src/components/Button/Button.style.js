const styles = {
  buttonStyle: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: '#fff',
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#007aff',
    marginLeft: 5,
    marginRight: 5
  },
  buttonTextStyle: {
    alignSelf: 'center',
    fontSize: 14,
    fontWeight: '600',
    paddingTop: 10,
    paddingBottom: 10,
    color: '#007aff'
  }
}

export default styles