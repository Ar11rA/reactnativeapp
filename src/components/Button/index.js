import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import buttonStyles from './Button.style'
const Button = (props) => {
  const { buttonStyle, buttonTextStyle } = buttonStyles
  return (
    <TouchableOpacity style={buttonStyle} onPress={() => props.onPress()}>
      <Text style={buttonTextStyle}>{props.text}</Text>
    </TouchableOpacity>  
  );
}

export default Button