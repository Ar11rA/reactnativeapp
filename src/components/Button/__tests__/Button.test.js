import 'react-native';
import React from 'react';
import Button from '../';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Button />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
