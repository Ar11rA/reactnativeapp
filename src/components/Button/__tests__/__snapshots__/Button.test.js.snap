// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#fff",
      "borderColor": "#007aff",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#007aff",
        "fontSize": 14,
        "fontWeight": "600",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    Buy Now
  </Text>
</View>
`;
